grammars/javascript.cson

```diff
@@ -17,8 +17,162 @@
 'name': 'JavaScript'
 'patterns': [
   {
+    'include': '#switch_statement'
+  }
+  {
+    # foo: function() { … }
+    'begin': '''(?x)
+      (\\b([a-zA-Z_?.$][\\w?.$]*))
+      \\s*(:)\\s*
+      \\b(?:(async)(?:\\s+))?
+      (function\\*?)
+      \\s*(\\*?)\\s*
```

Comment:
Wouldn't this combined with the previous line give us `function**`?  Only one asterisk is allowed.